dot/state/service.go

```diff
@@ -51,22 +51,28 @@ type Service struct {
 	// Below are for testing only.
 	BabeThresholdNumerator   uint64
 	BabeThresholdDenominator uint64
+
+	// Below are for state trie online pruner
+	retainBlocks int64
+	pruningMode  Pruning
 }
 
 // NewService create a new instance of Service
-func NewService(path string, lvl log.Lvl) *Service {
+func NewService(path string, lvl log.Lvl, pruningMode Pruning, retainBlocks int64) *Service {
```

Comment:
I think it would make more sense to have the `NewService` constructor take in a proposed `ServiceConfig` struct with an associated `PruningConfig` struct where we could put the `Mode` and `NumRetainedBlocks`.   We could have a `DefaultServiceConfig` defined that can be used by default.  We could then make the config a variadic parameter of the constructor.
```suggestion
type ServiceConfig{
   Path string
   LogLevel log.Lvl
   Pruning struct {
     Mode PruningMode
     NumRetainedBlocks int
   }
}
func NewService(config ...ServiceConfig) *Service {
```

Reply:
Modified it to take struct as param.